src/OrchardCore.Modules/OrchardCore.Users/Localization/Drivers/UserLocalizationDisplayDriver.cs

```diff
@@ -0,0 +1,67 @@
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.Extensions.Localization;
+using OrchardCore.DisplayManagement.Entities;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.DisplayManagement.ModelBinding;
+using OrchardCore.DisplayManagement.Views;
+using OrchardCore.Localization;
+using OrchardCore.Users.Localization.Models;
+using OrchardCore.Users.Localization.ViewModels;
+using OrchardCore.Users.Models;
+
+namespace OrchardCore.Users.Localization.Drivers;
+
+public class UserLocalizationDisplayDriver : SectionDisplayDriver<User, UserLocalizationSettings>
+{
+    private readonly ILocalizationService _localizationService;
+    protected readonly IStringLocalizer S;
+
+    public UserLocalizationDisplayDriver(
+        ILocalizationService localizationService,
+        IStringLocalizer<UserLocalizationDisplayDriver> localizer)
+    {
+        _localizationService = localizationService;
+        S = localizer;
+    }
+
+    public override Task<IDisplayResult> EditAsync(UserLocalizationSettings section, BuildEditorContext context)
+    {
+        return Task.FromResult<IDisplayResult>(Initialize<UserLocalizationViewModel>("UserCulture_Edit", async model =>
+        {
+            var supportedCultures = await _localizationService.GetSupportedCulturesAsync();
+
+            var cultureList = supportedCultures.Select(culture =>
+                new SelectListItem
+                {
+                    Text = CultureInfo.GetCultureInfo(culture).DisplayName + " (" + culture + ")",
+                    Value = culture
+                }).ToList();
+
+            cultureList.Insert(0, new SelectListItem() { Text = S["Use site's culture"], Value = "none" });
+
+            // If Invariant Culture is installed as a supported culture we bind it to a different culture code than String.Empty.
```

Comment:
Why?

Reply:
Just moving code to the driver to clean up the Razor view.

Reply:
Also now using "none" for that option so that a user can unset it's user culture.

Reply:
> Just moving code to the driver to clean up the Razor view.

That's what I did before then revert the commits :)

Reply:
True, I found out that the issue is that Model.Culture and SelectListItem.Selected was conflicting with each others. The issue was not in the Razor view.